### PR TITLE
Translate `library/random.po`

### LINKED ISSUE #155: Translate `library/random.po`

### PR DESCRIPTION
**Pull request by allen91wu**
resolve #155 

## Preview
image

### CODE REVIEW

File: library/random.po [hunk 1]
```diff
@@ -618,11 +715,14 @@ msgid ""
 "significance or `p-value <https://en.wikipedia.org/wiki/P-value>`_ of an "
 "observed difference between the effects of a drug versus a placebo::"
 msgstr ""
+"`重新取樣排列測試 <https://en.wikipedia.org/wiki/"
+"Resampling_(statistics)#Permutation_tests>`_ 的範例，來確定觀察到的藥物與安慰"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"Resampling_(statistics)#Permutation_tests>`_\\ 的範例，來確定觀察到的藥物與安慰"
```

File: library/random.po [hunk 2]
```diff
@@ -631,6 +731,9 @@ msgid ""
 "profile/295/>`_ on statistical analysis using just a few fundamental "
 "concepts including simulation, sampling, shuffling, and cross-validation."
 msgstr ""
+"`Statistics for Hackers <https://www.youtube.com/watch?v=Iq9DzN6mvYA>`_ 由 "
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"`Statistics for Hackers <https://www.youtube.com/watch?v=Iq9DzN6mvYA>`_ 是由 "
```

File: library/random.po [hunk 3]
```diff
@@ -618,11 +715,14 @@ msgid ""
 "significance or `p-value <https://en.wikipedia.org/wiki/P-value>`_ of an "
 "observed difference between the effects of a drug versus a placebo::"
 msgstr ""
+"`重新取樣排列測試 <https://en.wikipedia.org/wiki/"
+"Resampling_(statistics)#Permutation_tests>`_ 的範例，來確定觀察到的藥物與安慰"
+"劑之間差異的統計學意義或 `p值 <https://en.wikipedia.org/wiki/P-value>`_ ::"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"劑之間差異的統計學意義或 `p 值 <https://en.wikipedia.org/wiki/P-value>`_ ::"
```

File: library/random.po [hunk 4]
```diff
@@ -470,23 +536,30 @@ msgid ""
...
 "Gamma distribution.  (*Not* the gamma function!)  Conditions on the "
 "parameters are ``alpha > 0`` and ``beta > 0``."
 msgstr ""
+"Gamma（伽瑪）分佈。（*不是* Gamma 函式！） 參數的條件為 ``alpha > 0`` 和 "
+"``beta > 0``。"
 
 #: ../../library/random.rst:320
 msgid "The probability distribution function is::"
-msgstr ""
+msgstr "Probability distribution function（機率密度函數是）::"
```
Comment (mattwang44) [inline library/random.po]: Need to modify this based on [the instruction for rst syntax from readme](https://github.com/python/python-docs-zh-tw#rst). Please also check other source strings with `::`. 

```suggestion
msgstr ""
"Probability distribution function（機率密度函數是）：\n"
"\n"
"::"
```

Comment (mattwang44) [reply library/random.po]:
```suggestion
msgstr "Probability distribution function（機率密度函式）是 ::"
```

File: library/random.po [hunk 5]
```diff
@@ -470,23 +536,30 @@ msgid ""
...
+"指數分佈。*lambd* 為 1.0 除以所需的平均數。它應該不為零。（該參數將被稱為 "
+"\"lambda\"，但這是 Python 中的保留字）如果 *lambd* 為正，則回傳值的範圍從 0 "
+"到正無窮大；如果 *lambd* 為負，則回傳值的範圍從負無窮大到 0。"
 
 #: ../../library/random.rst:317
 msgid ""
 "Gamma distribution.  (*Not* the gamma function!)  Conditions on the "
 "parameters are ``alpha > 0`` and ``beta > 0``."
 msgstr ""
+"Gamma（伽瑪）分佈。（*不是* Gamma 函式！） 參數的條件為 ``alpha > 0`` 和 "
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"Gamma（伽瑪）分佈。（*不是* Gamma 函式！）參數的條件為 ``alpha > 0`` 和 "
```

File: library/random.po [hunk 6]
```diff
@@ -209,62 +209,74 @@ msgstr ""
 
 #: ../../library/random.rst:108
 msgid "Functions for bytes"
-msgstr ""
+msgstr "位元組函式"
 
 #: ../../library/random.rst:112
 msgid "Generate *n* random bytes."
-msgstr ""
+msgstr "產生 *n* 隨機位元組。"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
msgstr "產生 *n* 個隨機位元組。"
```

File: library/random.po [hunk 7]
```diff
@@ -209,62 +209,74 @@ msgstr ""
...
 msgid ""
 "This method should not be used for generating security tokens. Use :func:"
 "`secrets.token_bytes` instead."
 msgstr ""
+"此方法不應使用於產生安全性權杖 (Token)。請改用 :func:`secrets.token_bytes`。"
 
 #: ../../library/random.rst:121
 msgid "Functions for integers"
-msgstr ""
+msgstr "整數函式"
```
Comment (mattwang44) [inline library/random.po]: nit: `整數函式` is a bit vague imo (so does other titles, such as `序列函式` or `位元組函式`). It's fine if you prefer the current translations.

```suggestion
msgstr "回傳整數的函式"
```

File: library/random.po [hunk 8]
```diff
@@ -209,62 +209,74 @@ msgstr ""
...
 
 #: ../../library/random.rst:126
 msgid ""
 "Return a randomly selected element from ``range(start, stop, step)``.  This "
 "is equivalent to ``choice(range(start, stop, step))``, but doesn't actually "
 "build a range object."
 msgstr ""
+"從 ``range(start, stop, step)`` 中所回傳隨機選擇的元素。這等效於 "
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"從 ``range(start, stop, step)`` 中回傳一個隨機選擇的元素。這等效於 "
```

File: library/random.po [hunk 9]
```diff
@@ -209,62 +209,74 @@ msgstr ""
...
+"非整數類型到等效整數的自動轉換已被棄用。目前 ``randrange(10.0)`` 被無損轉換"
+"為 ``randrange(10)``。將來，這將會引發 :exc:`TypeError`。"
 
 #: ../../library/random.rst:143
 msgid ""
 "The exception raised for non-integral values such as ``randrange(10.5)`` or "
 "``randrange('10')`` will be changed from :exc:`ValueError` to :exc:"
 "`TypeError`."
 msgstr ""
+"對於非整數值，例如 ``randrange(10.5)`` 或 ``randrange('10')``，引發的異常將"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"對於非整數值，例如 ``randrange(10.5)`` 或 ``randrange('10')``，引發的例外將"
```

File: library/random.po [hunk 10]
```diff
@@ -209,62 +209,74 @@ msgstr ""
...
+"從 :exc:`ValueError` 改為 :exc:`TypeError`。"
 
 #: ../../library/random.rst:150
 msgid ""
 "Return a random integer *N* such that ``a <= N <= b``.  Alias for "
 "``randrange(a, b+1)``."
 msgstr ""
+"返回一個隨機整數 *N*，使得 ``a <= N <= b``。``randrange(a, b+1)`` 的別名。"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"回傳一個隨機整數 *N*，使得 ``a <= N <= b``。為 ``randrange(a, b+1)`` 的別名。"
```

File: library/random.po [hunk 11]
```diff
@@ -273,26 +285,33 @@ msgid ""
 "also provide it as an optional part of the API. When available, :meth:"
 "`getrandbits` enables :meth:`randrange` to handle arbitrarily large ranges."
 msgstr ""
+"回傳一個具有 *k* 隨機位的非負 Python 整數。此方法隨 MersenneTwister 與其他一"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"回傳一個具有 *k* 個隨機位元的非負 Python 整數。此方法隨 MersenneTwister 與其他一"
```

File: library/random.po [hunk 12]
```diff
@@ -273,26 +285,33 @@ msgid ""
...
 msgstr ""
+"從非空序列 *seq* 回傳一個隨機元素。如果 *seq* 為空，則引發 :exc:"
+"`IndexError`。"
 
 #: ../../library/random.rst:175
 msgid ""
 "Return a *k* sized list of elements chosen from the *population* with "
 "replacement. If the *population* is empty, raises :exc:`IndexError`."
 msgstr ""
+"回傳從 *population* 中選擇的大小為 *k* 的元素清單進行替換。如果 *population* "
```
Comment (mattwang44) [inline library/random.po]: - translate `list` as `串列` or keep the English content.
- what `sample with replacement` meant is `重置抽樣` or `放回取樣`.

```suggestion
"回傳從 *population* 中重置取樣出的一個大小為 *k* 的元素 list。如果 *population* "
```

File: library/random.po [hunk 13]
```diff
@@ -273,26 +285,33 @@ msgid ""
 "also provide it as an optional part of the API. When available, :meth:"
 "`getrandbits` enables :meth:`randrange` to handle arbitrarily large ranges."
 msgstr ""
+"回傳一個具有 *k* 隨機位的非負 Python 整數。此方法隨 MersenneTwister 與其他一"
+"些產生器一起提供，也可能將其作為 API 的可選部分提供。如果可用，\\ :meth:"
```
Comment (mattwang44) [inline library/random.po]: (1) supplied by MersenneTwister generator (2) provided as optional part by some other generators.

```suggestion
"回傳一個具有 *k* 隨機位的非負 Python 整數。此方法會隨 MersenneTwister 產生器一"
"起提供，一些其他的產生器也可能將其作為 API 的可選部分。如果可用，\\ :meth:"
```

File: library/random.po [hunk 14]
```diff
@@ -312,6 +336,9 @@ msgid ""
 "same length as the *population* sequence.  It is a :exc:`TypeError` to "
 "specify both *weights* and *cum_weights*."
 msgstr ""
+"如果既未指定 *weights* 也未指定 *cum_weights*，則以相等的機率進行選擇。如果提"
+"供了加權序列，則該序列的長度必須與 *population* 序列的長度相同。 它是一個 :"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"供了加權序列，則該序列的長度必須與 *population* 序列的長度相同。它是一個 :"
```

File: library/random.po [hunk 15]
```diff
@@ -321,6 +348,10 @@ msgid ""
 "to be non-negative and finite.  A :exc:`ValueError` is raised if all weights "
 "are zero."
 msgstr ""
+"*weights* 或 *cum_weights* 可以使用任何 :func:`random` 所回傳的 :class:"
+"`float` 值（包括整數、float 和分數，但不包括小數）互操作 (interoperates) 的數"
+"值類型。權重假定為非負數和有限權重。如果所有權重均為零，則引發 :exc:"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"值類型。權重假定為非負數和有限的。如果所有權重均為零，則引發 :exc:"
```

File: library/random.po [hunk 16]
```diff
@@ -321,6 +348,10 @@ msgid ""
 "to be non-negative and finite.  A :exc:`ValueError` is raised if all weights "
 "are zero."
 msgstr ""
+"*weights* 或 *cum_weights* 可以使用任何 :func:`random` 所回傳的 :class:"
+"`float` 值（包括整數、float 和分數，但不包括小數）互操作 (interoperates) 的數"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"*weights* 或 *cum_weights* 可以使用任何與 :func:`random` 所回傳的 :class:"
"`float` 值（包括整數、float 和分數，但不包括小數）交互操作 (interoperates) 的數"
```

File: library/random.po [hunk 17]
```diff
@@ -331,26 +362,34 @@ msgid ""
...
 
 #: ../../library/random.rst:207
 msgid "Raises a :exc:`ValueError` if all weights are zero."
-msgstr ""
+msgstr "如果所有權重均為零，則引發 :exc:`ValueError`。"
 
 #: ../../library/random.rst:213
 msgid "Shuffle the sequence *x* in place."
-msgstr ""
+msgstr "將序列 *x* 隨機打亂位置。"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
msgstr "將序列 *x* 原地 (in place) 隨機打亂位置。"
```

File: library/random.po [hunk 18]
```diff
@@ -331,26 +362,34 @@ msgid ""
...
 
 #: ../../library/random.rst:215
 msgid ""
 "The optional argument *random* is a 0-argument function returning a random "
 "float in [0.0, 1.0); by default, this is the function :func:`.random`."
 msgstr ""
+"選擇性引數 *random* 是一個 0 引數函式，預設回傳一個在 [0.0, 1.0) 的隨機 "
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"選擇性引數 *random* 是一個 0 引數函式，回傳一個在 [0.0, 1.0) 之間的隨機 "
```

File: library/random.po [hunk 19]
```diff
@@ -360,16 +399,20 @@ msgid ""
 "generated.  For example, a sequence of length 2080 is the largest that can "
 "fit within the period of the Mersenne Twister random number generator."
 msgstr ""
+"請注意，即使對於較小的 ``len(x)``，*x* 的排列總數也會快速成長到大於大多數隨機"
+"數產生器的週期。這意味著長序列的大多數排列永遠無法產生。例如，長度為 2080 的"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"請注意，即使對於較小的 ``len(x)``，*x* 的置換總數也會快速成長到大於大多數隨機"
"數產生器的週期。這意味著長序列的大多數置換永遠無法產生。例如，長度為 2080 的"
```

File: library/random.po [hunk 20]
```diff
@@ -360,16 +399,20 @@ msgid ""
...
 msgid "The optional parameter *random*."
-msgstr ""
+msgstr "選擇性參數 *random*。"
 
 #: ../../library/random.rst:233
 msgid ""
 "Return a *k* length list of unique elements chosen from the population "
 "sequence or set. Used for random sampling without replacement."
 msgstr ""
+"回傳從母體序列或集合中選擇的唯一元素的 *k* 長度清單。用於隨機取樣，無需替換。"
```
Comment (mattwang44) [inline library/random.po]: - what `sample without replacement` meant is `不重置抽樣` or `不放回取樣`.
```suggestion
"回傳從母體序列或集合中選擇出的一個包含獨特元素、長度為 *k* 的 list。用於不重置的隨機取樣。"
```

File: library/random.po [hunk 21]
```diff
@@ -610,6 +704,9 @@ msgid ""
 "Bootstrapping_(statistics)>`_ using resampling with replacement to estimate "
 "a confidence interval for the mean of a sample::"
 msgstr ""
+"`統計 bootstrapping（自助法） <https://en.wikipedia.org/wiki/"
+"Bootstrapping_(statistics)>`_ 的範例，使用替換重新取樣來估計樣本平均數的信賴"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"Bootstrapping_(statistics)>`_\\ 的範例，使用有重置的重新取樣來估計樣本平均數的信賴"
```

File: library/random.po [hunk 22]
```diff
@@ -518,33 +596,39 @@ msgid ""
...
 msgid "Alternative Generator"
-msgstr ""
+msgstr "替代產生器"
 
 #: ../../library/random.rst:378
 msgid ""
 "Class that implements the default pseudo-random number generator used by "
 "the :mod:`random` module."
-msgstr ""
+msgstr "實現 :mod:`random` 模組使用的預設偽隨機數產生器的 Class。"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
msgstr "實現 :mod:`random` 模組使用的預設偽隨機數產生器的 class。"
```

File: library/random.po [hunk 23]
```diff
@@ -379,13 +422,18 @@ msgid ""
 "winners (the sample) to be partitioned into grand prize and second place "
 "winners (the subslices)."
 msgstr ""
+"回傳包含母體元素的新清單，同時保持原始母體不變。 產生的清單按選擇順序排列，因"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"回傳包含母體元素的新清單，同時保持原始母體不變。產生的清單按選擇順序排列，因"
```

File: library/random.po [hunk 24]
```diff
@@ -419,10 +472,12 @@ msgid ""
 "`list` or :class:`tuple`, preferably in a deterministic order so that the "
 "sample is reproducible."
 msgstr ""
+"將來，*母體* 必須是一個序列。不再支援 :class:`set` 的實例。必須先將集合轉換"
```
Comment (mattwang44) [inline library/random.po]: - The argument name should not be translated.

```suggestion
"將來，*population* 必須是一個序列。不再支援 :class:`set` 的實例。必須先將集合轉換"
```

File: library/random.po [hunk 25]
```diff
@@ -419,10 +472,12 @@ msgid ""
...
+"為 :class:`list` 或 :class:`tuple`，最好是按確定性順序，以便樣本是可復現的。"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"為 :class:`list` 或 :class:`tuple`，最好是按確定性順序，以便取樣是可復現的。"
```

File: library/random.po [hunk 26]
```diff
@@ -431,22 +486,28 @@ msgid ""
 "distribution's equation, as used in common mathematical practice; most of "
 "these equations can be found in any statistics text."
 msgstr ""
+"以下函式產生特定的實數分佈。函式參數以分佈方程中的對應變量命名，如常見的數學"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"以下函式產生特定的實數分佈。函式參數以分佈方程中的對應變數命名，如常見的數學"
```

File: library/random.po [hunk 27]
```diff
@@ -431,22 +486,28 @@ msgid ""
...
-msgstr ""
+msgstr "回傳範圍 [0.0, 1.0) 中的下一個隨機浮點數。"
 
 #: ../../library/random.rst:285
 msgid ""
 "Return a random floating point number *N* such that ``a <= N <= b`` for ``a "
 "<= b`` and ``b <= N <= a`` for ``b < a``."
 msgstr ""
+"回傳一個隨機浮點數 *N*，使得 ``a <= N <= b`` 為 ``a <= b``、``b <= N <= a`` "
+"為 ``b < a``。"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"回傳一個隨機浮點數 *N*，當 ``a <= b`` 時確保 N 為 ``a <= N <= b`` 、``b < a`` "
"時確保 N 為 ``b <= N <= a``。"
```

File: library/random.po [hunk 28]
```diff
@@ -567,24 +655,30 @@ msgid ""
 "should be reproducible from run to run as long as multiple threads are not "
 "running."
 msgstr ""
+"有時，能夠重現偽隨機數產生器給出的序列很有用。只要多線程未運行，透過重複使用"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"有時，能夠重現偽隨機數產生器給出的序列很有用。只要多執行緒未運行，透過重複使用"
```

File: library/random.po [hunk 29]
```diff
@@ -496,6 +569,9 @@ msgid ""
 "random number generator. 2) Put locks around all calls. 3) Use the slower, "
 "but thread-safe :func:`normalvariate` function instead."
 msgstr ""
+"多執行緒須注意：當兩個線程同時呼叫此函式時，它們可能會收到相同的傳回值。 這可"
```
Comment (mattwang44) [inline library/random.po]: - Unify the translation of `thread` as `執行緒`.

```suggestion
"多執行緒須注意：當兩個執行緒同時呼叫此函式時，它們可能會收到相同的傳回值。這可"
```

File: library/random.po [hunk 30]
```diff
@@ -504,12 +580,14 @@ msgid ""
 "deviation *sigma*.  *mu* can have any value, and *sigma* must be greater "
 "than zero."
 msgstr ""
+"對數常態分佈。如果你取此分佈的自然對數，您將獲得一個具有平均數 *mu* 和標準差 "
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"對數常態分佈。如果你取此分佈的自然對數，你將獲得一個具有平均數 *mu* 和標準差 "
```

File: library/random.po [hunk 31]
```diff
@@ -504,12 +580,14 @@ msgid ""
...
+"*sigma* 的常態分佈。*mu* 可以有任何值，並且 *sigma* 必須大於零。"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"*sigma* 的常態分佈。*mu* 可以為任何值，並且 *sigma* 必須大於零。"
```

File: library/random.po [hunk 32]
```diff
@@ -555,10 +639,14 @@ msgid ""
 "ignored. The :meth:`getstate` and :meth:`setstate` methods raise :exc:"
 "`NotImplementedError` if called."
 msgstr ""
+"使用 :func:`os.urandom` 函式從作業系統提供的來源產生隨機數的 Class。並非在所"
+"有系統上都可用。不依賴於軟體狀態，並且序列不可復現。因此 :meth:`seed` 方法沒"
+"有效果，而被忽略。如果呼叫 :meth:`getstate` 和 :meth:`setstate` 方法會引發 :"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"有效果且被忽略。如果呼叫 :meth:`getstate` 和 :meth:`setstate` 方法會引發 :"
```

File: library/random.po [hunk 33]
```diff
@@ -567,24 +655,30 @@ msgid ""
...
 msgstr ""
+"大多數隨機 module 的演算法和 seed 設定函數在 Python 版本中可能會發生變化，但"
+"可以保證兩個方面不會改變："
 
 #: ../../library/random.rst:406
 msgid ""
 "If a new seeding method is added, then a backward compatible seeder will be "
 "offered."
-msgstr ""
+msgstr "如果增加了新的播種方法，則將提供向後相容的播種器。"
```
Comment (mattwang44) [inline library/random.po]: - Unify the translation for `seeding function`.
- Add original content for `seeder`.

```suggestion
msgstr "如果增加了新的 seed 設定函數，則將提供向後相容的播種器 (seeder)。"
```

File: library/random.po [hunk 34]
```diff
@@ -618,11 +715,14 @@ msgid ""
...
 msgstr ""
+"`重新取樣排列測試 <https://en.wikipedia.org/wiki/"
+"Resampling_(statistics)#Permutation_tests>`_ 的範例，來確定觀察到的藥物與安慰"
+"劑之間差異的統計學意義或 `p值 <https://en.wikipedia.org/wiki/P-value>`_ ::"
 
 #: ../../library/random.rst:513
 msgid ""
 "Simulation of arrival times and service deliveries for a multiserver queue::"
-msgstr ""
+msgstr "模擬多伺服器序列的到達時間與服務交付 ::"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
msgstr "模擬多伺服器佇列 (queue) 的到達時間與服務交付 ::"
```

File: library/random.po [hunk 35]
```diff
@@ -640,6 +743,10 @@ msgid ""
 "distributions provided by this module (gauss, uniform, sample, betavariate, "
 "choice, triangular, and randrange)."
 msgstr ""
+"`Economics Simulation <http://nbviewer.jupyter.org/url/norvig.com/ipython/"
+"Economics.ipynb>`_ 由 `Peter Norvig <http://norvig.com/bio.html>`_ 對市場進行"
+"了模擬，顯示了該模組提供的許多工具和分佈（高斯、均勻、樣本、 beta 變數、選"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"Economics.ipynb>`_ 是由 `Peter Norvig <http://norvig.com/bio.html>`_ 對市場進行"
"的模擬，顯示了該模組提供的許多工具和分佈（高斯、均勻、樣本、 beta 變數、選"
```

File: library/random.po [hunk 36]
```diff
@@ -649,10 +756,14 @@ msgid ""
 "theory, how to write simulations, and how to perform data analysis using "
 "Python."
 msgstr ""
+"`機率的具體介紹（使用Python） <http://nbviewer.jupyter.org/url/norvig.com/"
+"ipython/Probability.ipynb>`_ 為 `Peter Norvig <http://norvig.com/bio.html>`_ "
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"ipython/Probability.ipynb>`_\\ 為 `Peter Norvig <http://norvig.com/bio.html>`_ "
```

File: library/random.po [hunk 37]
```diff
@@ -649,10 +756,14 @@ msgid ""
...
+"的教學課程，涵蓋了機率理論的基礎知識，如何模擬以及如何使用 Python 執行數據分"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"的教學課程，涵蓋了機率理論的基礎知識與如何模擬以及使用 Python 執行數據分"
```

File: library/random.po [hunk 38]
```diff
@@ -649,10 +756,14 @@ msgid ""
...
+"析。"
 
 #: ../../library/random.rst:563
 msgid "Recipes"
-msgstr ""
+msgstr "程式庫"
```
Comment (mattwang44) [inline library/random.po]: pretty tricky to translate this term... I'm open to any better idea.

```suggestion
msgstr "使用方案"
```

File: library/random.po [hunk 39]
```diff
@@ -671,12 +785,17 @@ msgid ""
 "geometric distribution where exponents smaller than *-53* occur half as "
 "often as the next larger exponent."
 msgstr ""
+"以下範例採用不同的方法。間隔中的所有 float 都是可能的選擇。尾數來自 *2⁵² ≤ 尾"
+"數< 2⁵³* 範圍內的整數均勻分佈。指數來自幾何分佈，其中小於 *-53* 的指數的出現"
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"數 < 2⁵³* 範圍內的整數均勻分佈。指數來自幾何分佈，其中小於 *-53* 的指數的出現"
```

File: library/random.po [hunk 40]
```diff
@@ -671,12 +785,17 @@ msgid ""
...
+"頻率是下一個較大指數的一半。"
 
 #: ../../library/random.rst:593
 msgid ""
 "All :ref:`real valued distributions <real-valued-distributions>` in the "
 "class will use the new method::"
 msgstr ""
+"Class 中的所有 :ref:`real valued distribution <real-valued-distributions>` 都"
```
Comment (mattwang44) [inline library/random.po]: The name of the hyperlink can be translated.

```suggestion
"Class 中的所有\\ :ref:`實數分佈 <real-valued-distributions>`\\ 都"
```

File: library/random.po [hunk 41]
```diff
@@ -686,6 +805,10 @@ msgid ""
 "Python float.  (The value 2⁻¹⁰⁷⁴ is the smallest positive unnormalized float "
 "and is equal to ``math.ulp(0.0)``.)"
 msgstr ""
+"該範例在概念上等效於一種演算法，該演算法從 *0.0 ≤ x < 1.0* 範圍內 2⁻¹⁰⁷⁴ 的所"
+"有倍數中進行選擇。這些數字都是均勻分佈的，但大多數必須向下捨入到最接近的可表"
+"示的 Python float。（ 2⁻¹⁰⁷⁴ 是最小為正的非正規化 float，等於 ``math."
```
Comment (mattwang44) [inline library/random.po]:
```suggestion
"示的 Python float。（2⁻¹⁰⁷⁴ 是最小為正的非正規化 float，等於 ``math."
```

File: library/random.po (hunk 33 above)
Comment (allen91wu) [reply library/random.po]: `seeding method` translate to `seed 設定方法` would be better?

Comment (mattwang44) [reply library/random.po]: imo, `seed 設定函數` is better. If it's too troublesome, we can just leave it untranslated :P

**Comment by allen91wu** [comment]
## New Preview
image

**Comment by allen91wu** [comment]
## New Preview
image

File: library/random.po (hunk 4 above)
Comment (mattwang44) [reply library/random.po]: For the rst syntax issue, you've missed the colons (`：`) in the translation in the new commit.

File: library/random.po (hunk 7 above)
Comment (mattwang44) [reply library/random.po]: I've noticed you've changed this line in the new commit. You may need to also change all other titles in this format (e.g. `Functions for sequences` -> `回傳序列的函式`)

File: library/random.po [hunk 42]
```diff
@@ -567,24 +660,30 @@ msgid ""
...
 msgstr ""
+"有時，能夠重現偽隨機數產生器給出的序列很有用。只要多執行緒未運行，透過重複使"
+"用種子值，同一序列就應該可以被復現。"
 
 #: ../../library/random.rst:403
 msgid ""
 "Most of the random module's algorithms and seeding functions are subject to "
 "change across Python versions, but two aspects are guaranteed not to change:"
 msgstr ""
+"大多數隨機 module 的演算法和 seed 設定函數在 Python 版本中可能會發生變化，但"
```
Comment (mattwang44) [inline library/random.po]: I missed this during the last review: let's use `函式` for `function` (I can find 7 `函數` in the latest version of this PR).
```suggestion
"大多數隨機 module 的演算法和 seed 設定函式在 Python 版本中可能會發生變化，但"
```

File: library/random.po [hunk 43]
```diff
@@ -618,11 +722,20 @@ msgid ""
 "significance or `p-value <https://en.wikipedia.org/wiki/P-value>`_ of an "
 "observed difference between the effects of a drug versus a placebo::"
 msgstr ""
+"`重新取樣排列測試 <https://en.wikipedia.org/wiki/"
+"Resampling_(statistics)#Permutation_tests>`_\\ 的範例，來確定觀察到的藥物與安"
+"慰劑之間差異的統計學意義或 `p 值 <https://en.wikipedia.org/wiki/P-value>`_"
+"\\：\n"
```
Comment (mattwang44) [inline library/random.po]: a space here?
```suggestion
"\\ ：\n"
```

**Review by mattwang44 — COMMENTED** [review]
two tiny issues left. I believe we can merge this after they're solved!

**Comment by allen91wu** [comment]
In the latest commit, I have also fixed the other translation typo from `函數` to `函式` in this page.
Please have a look. Thanks!

**Review by mattwang44 — APPROVED** [review]
Nice work! 辛苦了

**Pull request merged**